docs/core/diagnostics/distributed-tracing.md

```diff
@@ -0,0 +1,229 @@
+---
+# .NET Distributed Tracing
+
+Distributed tracing is the way to publish and observe the tracing data in a distributed system.
+.NET Framework and .NET Core has been supporting tracing through the <xref:System.Diagnostics> APIs.
+
+- <xref:System.Diagnostics.Activity?displayProperty=nameWithType> class which allows storing and accessing diagnostics context and consuming it with logging system.
+- <xref:System.Diagnostics.DiagnosticSource?displayProperty=nameWithType> that allows code to be instrumented for production-time logging of rich data payloads for consumption within the process that was instrumented.
+
+Here is an example shows how to publish tracing data from the HTTP incoming requests:
+
+```csharp
+   public void OnIncomingRequest(DiagnosticListener httpListener, HttpContext context)
+   {
+       if (httpListener.IsEnabled("Http_In"))
+       {
+           var activity = new Activity("Http_In");
+
+           //add tags, baggage, etc.
+           activity.SetParentId(context.Request.headers["Request-id"])
+           foreach (var pair in context.Request.Headers["Correlation-Context"])
+           {
+               var baggageItem = NameValueHeaderValue.Parse(pair);
+               activity.AddBaggage(baggageItem.Key, baggageItem.Value);
+           }
+           httpListener.StartActivity(activity, new { context });
+           try
+           {
+               //process request ...
+           }
+           finally
+           {
+               //stop activity
+               httpListener.StopActivity(activity, new {context} );
+           }
+       }
+   }
+```
+
+Here is example for how to listen to the Activity events:
+
+```csharp
+    DiagnosticListener.AllListeners.Subscribe(delegate (DiagnosticListener listener)
+    {
+        if (listener.Name == "MyActivitySource")
+        {
+            listener.Subscribe(delegate (KeyValuePair<string, object> value)
+            {
+                if (value.Key.EndsWith("Start", StringComparison.Ordinal))
+                    LogActivityStart();
+                else if (value.Key.EndsWith("Stop", StringComparison.Ordinal))
+                    LogActivityStop();
+            });
+        }
+    }
+```
+
+.NET 5.0 has extended the capability of the distributed tracing to allow the [OpenTelemetry](https://opentelemetry.io/) tracing scenarios, added Sampling capabilities, simplified the tracing coding pattern, and made the listening to the Activity events easier and flexible.
+
+> [!NOTE]
+> To access all added .NET 5.0 capabilities, ensure referencing the [System.Diagnostics.DiagnosticSource](https://www.nuget.org/packages/System.Diagnostics.DiagnosticSource/) NuGet package version 5.0 or later. This package can be used in libraries and apps targeting any supported version of the .NET Framework, .NET Core, and .NET Standard. If targeting .NET 5.0 or later, no need to manually reference the package as it is included in the shared library installed with the .NET SDK.
```

Comment:
To access all added .NET 5.0 capabilities, ensure **your application references** the

Reply:
This may need to mention `application or library` I guess.